From d2e0c55b38cf6843ad1dde536ad281d6fb3dca27 Mon Sep 17 00:00:00 2001
From: Zi Shen Lim <zlim@netlogicmicro.com>
Date: Tue, 10 Jan 2012 12:03:54 -0800
Subject: [PATCH 325/565] XLR: undo overzealous purge

XLR: undo overzealous purge that broke XLP build

Based on Broadcom SDK 2.3.

Signed-off-by: Zi Shen Lim <zlim@netlogicmicro.com>
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 arch/mips/include/asm/netlogic/iomap.h | 30 ++++++++++++++++++++++++++++++
 1 file changed, 30 insertions(+)

diff --git a/arch/mips/include/asm/netlogic/iomap.h b/arch/mips/include/asm/netlogic/iomap.h
index 772dac1..8a1418e 100644
--- a/arch/mips/include/asm/netlogic/iomap.h
+++ b/arch/mips/include/asm/netlogic/iomap.h
@@ -36,6 +36,36 @@ THE POSSIBILITY OF SUCH DAMAGE.
 #define NETLOGIC_IO_UART_0_OFFSET          0x30100
 #define NETLOGIC_IO_UART_1_OFFSET          0x31100
 #define NETLOGIC_IO_SYS_OFFSET             0x35100
+
+#define NETLOGIC_IO_USB_1_OFFSET           0x25000
+#define NETLOGIC_IO_GPIO_OFFSET            0x18000
+
+#ifndef __ASSEMBLY__
+
+#include <linux/types.h>
+#include <asm/byteorder.h>
+
+typedef volatile __u32 nlm_reg_t;
+extern unsigned long netlogic_io_base;
+
+#define netlogic_io_mmio(offset) ((nlm_reg_t *)(netlogic_io_base+(offset)))
+
+#ifdef CONFIG_CPU_LITTLE_ENDIAN
+
+#define netlogic_read_reg(base, offset) ((base)[(offset)])
+#define netlogic_write_reg(base, offset, value) ((base)[(offset)] = (value))
+
+#else
+
+#define netlogic_read_reg(base, offset) (be32_to_cpu((base)[(offset)]))
+#define netlogic_write_reg(base, offset, value) ((base)[(offset)] = cpu_to_be32((value)))
+
+#endif
+
+extern void on_chip_init(void);
+
+#endif /* __ASSEMBLY__ */
+
 #endif /* CONFIG_NLM_XLP */
 
 #define NETLOGIC_IO_SIZE                   0x1000
-- 
1.8.4.93.g57e4c17

